moto/ram/models.py

```diff
@@ -0,0 +1,257 @@
+            )
+            if match:
+                roots = self.organizations_backend.list_roots()
+                root_id = next(
+                    (
+                        root["Id"]
+                        for root in roots["Roots"]
+                        if root["Name"] == "Root" and match.group(1) in root["Arn"]
+                    ),
+                    None,
+                )
+
+                if root_id:
+                    ous = self.organizations_backend.list_organizational_units_for_parent(
+                        ParentId=root_id
+                    )
+                    if any(principal == ou["Arn"] for ou in ous["OrganizationalUnits"]):
+                        continue
+
+                raise UnknownResourceException(
+                    "OrganizationalUnit {} in unknown organization could not be found.".format(
+                        match.group(2)
+                    )
+                )
+
+            if not re.match(r"^\d{12}$", principal):
+                raise InvalidParameterException(
+                    "Principal ID {} is malformed. "
+                    "Verify the ID and try again.".format(principal)
+                )
+
+        for principal in principals:
+            self.principals.append(principal)
+
+    def add_resources(self, resource_arns):
+        for resource in resource_arns:
+            match = re.search(
+                r"^arn:aws:[a-z0-9-]+:[a-z0-9-]*:[0-9]{12}:([a-z-]+)[/:].*$", resource
+            )
+            if not match:
+                raise MalformedArnException(
+                    "The specified resource ARN {} is not valid. "
+                    "Verify the ARN and try again.".format(resource)
+                )
+
+            if match.group(1) not in self.SHAREABLE_RESOURCES:
+                raise MalformedArnException(
+                    "You cannot share the selected resource type."
+                )
+
+        for resource in resource_arns:
+            self.resource_arns.append(resource)
+
+    def delete(self):
+        self.last_updated_time = datetime.utcnow()
+        self.status = "DELETED"
+
+    def describe(self):
+        return {
+            "allowExternalPrincipals": self.allow_external_principals,
+            "creationTime": unix_time(self.creation_time),
+            "featureSet": self.feature_set,
+            "lastUpdatedTime": unix_time(self.last_updated_time),
+            "name": self.name,
+            "owningAccountId": self.owning_account_id,
+            "resourceShareArn": self.arn,
+            "status": self.status,
+        }
+
+    def update(self, **kwargs):
+        self.allow_external_principals = kwargs.get(
+            "allowExternalPrincipals", self.allow_external_principals
+        )
+        self.last_updated_time = datetime.utcnow()
+        self.name = kwargs.get("name", self.name)
+
+    def _create_arn(self):
+        id = "{0}-{1}-{2}-{3}-{4}".format(
```

Comment:
This looks like a UUID - does it make sense to that instead? (`from uuid import uuid4`?)

Reply:
Good eye! Somehow didn't realize it 😄